File: Iniciante/PHP/exercise_1045.php
<?php

$valores = readline();

$valores = explode(" ", $valores);

$valores = array_map(function ($valor) { return floatval($valor);}, $valores);

rsort($valores);

switch (true) {
    case ($valores[0] >= ($valores[1]+$valores[2])):
        print_r("NAO FORMA TRIANGULO\n");
        break;
    case (pow($valores[0], 2) == (pow($valores[1], 2)+pow($valores[2], 2))):
        print_r("TRIANGULO RETANGULO\n");
        break;
    case (pow($valores[0], 2) > (pow($valores[1], 2)+pow($valores[2], 2))):
        print_r("TRIANGULO OBTUSANGULO\n");
        break;
    case (pow($valores[0], 2) < (pow($valores[1], 2)+pow($valores[2], 2))):
        print_r("TRIANGULO ACUTANGULO\n");
        break;
}

if ($valores[0] == $valores[1] && $valores[1] == $valores[2]) {
    print_r("TRIANGULO EQUILATERO\n");
} else if ($valores[0] == $valores[1] || $valores[1] == $valores[2] || $valores[0] == $valores[2]) {
    print_r("TRIANGULO ISOSCELES\n");
}

//Concluído!

File: Iniciante/Python/exercise_1045.py
def isNotTriangle(valorA, valorB, valorC):
    return valorA >= (valorB+valorC)

def isRightTriangle(valorA, valorB, valorC):
    return valorA**2 == (valorB**2+valorC**2)

def isObtusangleTriangle(valorA, valorB, valorC):
    return valorA**2 > (valorB**2+valorC**2)

def isAcuteTriangle(valorA, valorB, valorC):
    return valorA ** 2 < (valorB ** 2 + valorC ** 2)

def isEquilateralTriangle(valorA, valorB, valorC):
    return valorA == valorB and valorB == valorC

def isIsoscelesTriangle(valorA, valorB, valorC):
    return valorA == valorB or valorB == valorC or valorA == valorC

valores = [float(x) for x in input().split()]
valores.sort(reverse=True)
valorA, valorB, valorC = valores

if isNotTriangle(valorA, valorB, valorC):
    print('NAO FORMA TRIANGULO')
elif isRightTriangle(valorA, valorB, valorC):
    print('TRIANGULO RETANGULO')
elif isObtusangleTriangle(valorA, valorB, valorC):
    print('TRIANGULO OBTUSANGULO')
elif isAcuteTriangle(valorA, valorB, valorC):
    print('TRIANGULO ACUTANGULO')

if isEquilateralTriangle(valorA, valorB, valorC):
    print('TRIANGULO EQUILATERO')
elif isIsoscelesTriangle(valorA, valorB, valorC):
    print('TRIANGULO ISOSCELES')